internal/runtime/internal/controller/loader.go

```diff
@@ -829,12 +840,18 @@ func (l *Loader) evaluate(logger log.Logger, bn BlockNode) error {
 func (l *Loader) postEvaluate(logger log.Logger, bn BlockNode, err error) error {
 	switch c := bn.(type) {
 	case ComponentNode:
-		// Always update the cache both the arguments and exports, since both might
-		// change when a component gets re-evaluated. We also want to cache the arguments and exports in case of an error
-		l.cache.CacheArguments(c.ID(), c.Arguments())
-		l.cache.CacheExports(c.ID(), c.Exports())
+		// Always update the cache both the exports, since that it might change when a component gets re-evaluated.
+		// We also want to cache it in case of an error
+		err2 := l.cache.CacheExports(c.ID(), c.Exports())
+		if err2 != nil {
+			level.Error(logger).Log("msg", "failed to cache exports after evaluation", "err", err2)
+			// Don't mask the previous evaluation error.
```

Comment:
yeah not a big fan of this, I think that when there is an evaluation error in modules, this checks whether a mandatory argument is missing. In this case it shadows the evaluation error because it might be just a consequence of the fact that the arguments are missing. I think that there should be better ways to do this but I don't want to modify it in this PR, I can add comments though